db/flush_job.cc

```diff
@@ -130,6 +130,7 @@ FlushJob::FlushJob(
 }
 
 FlushJob::~FlushJob() {
+  io_status_.PermitUncheckedError();
```

Comment:
Can we add assert ```io_status()```  is ok in flush_job_test.cc instead of ```PermitUncheckedError()```?

Reply:
I actually think how it is coded is correct and it should not be an assert.  If one simply creates a FlushJob and does nothing with it, then the assert would be triggered.

Reply:
@akankshamahajan15 I agree with what Mark said.